### PR TITLE
fix: use EthTraceMapper to mapper geth trace

### PR DESCRIPTION
**Pull request by ddl-hust**
`{"type": "trace", "transaction_index": 0, "from_address": "0x4200000000000000000000000000000000000006", "to_address": "0x10b589852cadcec648a687b94f17142954fcab3f", "value": 1063798636750241079, "input": "0x", "output": null, "trace_type": "call", "call_type": "call", "reward_type": null, "gas": null, "gas_used": null, "subtraces": 0, "trace_address": [0], "error": null, "status": 1, "block_number": 170061, "trace_id": "call_170061_1", "trace_index": 1, "transaction_hash": "0x17ed6bb4317c1490e36fb0f9e63694b5883bd7a14dd1f0484dbc01630edbc18e", "item_id": "trace_call_170061_1"}`

### CODE REVIEW

**Comment by ddl-hust** [comment]
still have some bug for tx hash

File: ethereumetl/jobs/export_geth_traces_job.py [hunk 1]
```diff
@@ -72,8 +76,15 @@ def _export_batch(self, block_number_batch):
                 'block_number': block_number,
                 'transaction_traces': [tx_trace.get('result') for tx_trace in result],
             })
-
-            self.item_exporter.export_item(self.geth_trace_mapper.geth_trace_to_dict(geth_trace))
+            # fix when transaction_traces is empty cause trace.from_address = to_normalized_address(tx_trace.get('from'))
+            # AttributeError: 'NoneType' object has no attribute 'get'
+            if len(geth_trace.transaction_traces) == 0:
```
Comment (ddl-hust) [inline ethereumetl/jobs/export_geth_traces_job.py:81]: 这里不需要了，我去掉下

File: .github/workflows/release-image-ecr.yaml [hunk 2]
```diff
@@ -4,11 +4,18 @@ on:
   push:
     branches:
```
Comment (TalkWIthKeyboard) [inline .github/workflows/release-image-ecr.yaml:5]: Are the conditions must be satisfied together?

File: ethereumetl/streaming/eth_streamer_adapter.py [hunk 3]
```diff
@@ -88,7 +95,23 @@ def export_all(self, start_block, end_block):
...
             if EntityType.TOKEN in self.entity_types else []
-
+        
+        # enrich trace 
+        if self.node_client == "geth":
+            # 1) firsh enrich to get transactions_hash
+            enriched_traces_by_blocks_transactions = enrich_traces_with_blocks_transactions(blocks, traces, transactions)
+            # 2) 补充trace_status
+            # dict -> trace object
+            trs = []
```
Comment (TalkWIthKeyboard) [inline ethereumetl/streaming/eth_streamer_adapter.py]: `trs = [trace_mapper.dict_to_trace(tr) for tr in enriched_traces_by_blocks_transactions]`

File: .github/workflows/release-image-ecr.yaml (hunk 2 above)
Comment (ddl-hust) [reply .github/workflows/release-image-ecr.yaml:5]: no,It's OR logical

File: ethereumetl/streaming/eth_streamer_adapter.py [hunk 4]
```diff
@@ -88,7 +95,23 @@ def export_all(self, start_block, end_block):
             if EntityType.CONTRACT in self.entity_types else []
         enriched_tokens = enrich_tokens(blocks, tokens) \
             if EntityType.TOKEN in self.entity_types else []
-
+        
+        # enrich trace 
+        if self.node_client == "geth":
```
Comment (TalkWIthKeyboard) [inline ethereumetl/streaming/eth_streamer_adapter.py]: It would be better if we would overwrite `enrich_traces` method that adds the logic to it.

Comment (TalkWIthKeyboard) [reply ethereumetl/streaming/eth_streamer_adapter.py]: 这里有一个小经验，我们需要时刻注意代码的内聚程度（代码的墒是否在降低），最好能找到最小的代码影响部分，然后去改那个地方。

Comment (ddl-hust) [reply ethereumetl/streaming/eth_streamer_adapter.py]: learned a lot

**Review by TalkWIthKeyboard — APPROVED** [review]

**Pull request merged**